scala/integration/maven/src/org/jetbrains/plugins/scala/project/maven/ScalaMavenImporter.scala

```diff
@@ -111,7 +111,9 @@ final class ScalaMavenImporter extends MavenApplicableConfigurator(PluginGroupId
     val implicitScalaLibraryInfo = Option(mavenProject.getCachedValue(MavenImplicitScalaLibraryInfo))
     implicitScalaLibraryInfo.map { info =>
       val vfUrlManager = WorkspaceModel.getInstance(project).getVirtualFileUrlManager
-      val jarUrl = vfUrlManager.getOrCreateFromUrl(s"jar://${info.path}!/")
+      // Use proper URI encoding to handle special characters in JAR paths
+      val jarUri = Paths.get(info.path).toUri.toString
+      val jarUrl = vfUrlManager.getOrCreateFromUrl(s"jar://$jarUri!/")
```

Comment:
Making an uri from uri ? `jar://file://`? 

+ how is it related to [SCL-24273](https://youtrack.jetbrains.com/issue/SCL-24273)?